src/vehicle.cpp

```diff
@@ -5182,11 +5182,14 @@ bool vehicle::fire_turret (int p, bool /* burst */ )
         if( liquid_fuel < 1 || charges < 1 ) {
             return false;
         }
-        itype *am_type = item::find_type( amt );
+        itype *am_type = item::find_type( default_ammo( amt ) );
+        itype *fake;
+        bool fake_ammo = false;
         if( !am_type->ammo ) {
-            // Let non-ammo-typed ammo be used too
-            // 0 in all stats, depends on gun to provide everything
-            am_type = item::find_type( "fake_ammo" );
+            // Alter ammo type temporarily
+            fake = item::find_type( "fake_ammo" );
+            std::swap( am_type->ammo, fake->ammo );
```

Comment:
While this an ingenious solution, I suggest removing the fake ammo completely. People should implement a proper ammo type for their guns instead of relying on this.